ui/model/model.py

```diff
@@ -149,6 +150,21 @@ def undo(self) -> GameHistoryItem:
 
         return next_item
 
+    def load_game_state(self, starting_layout: BoardLayout, game_history: GameHistory):
+        """
+        Loads game state from a given layout and history.
+        :param starting_layout: a BoardLayout
+        :param game_history: a GameHistory
+        """
+        board = Board.create_from_data(starting_layout.value)
+        for item in game_history:
+            board.apply_move(item.move)
+
+        self.game.set_board(board)
+        self.game.set_turn(game_history.infer_player_turn())
+        self.config.layout = starting_layout
+        self.history = game_history
```

Comment:
Effectively identical to the undo reconstruction logic, just stripped down a little bit for this particular use case (could merge the two fns if we cared)